app/controllers/users_controller.rb

```diff
@@ -0,0 +1,50 @@
+class UsersController < ApplicationController
+  def login_form
+    @user = User.new
+  end
+
+  def login
+    name = params[:user][:name]
+    user = User.find_by(name: name)
+    if user.nil?
+      user = User.create(name: name)
+      session[:user_id] = user.id
+      flash[:success] = "Successfully created new user #{user.name} with ID #{user.id}"
+    else
+      session[:user_id] = user.id
+      flash[:success] = "Successfully logged in as existing user #{user.name}"
+    end
+
+    if user.id
+      redirect_to root_path
+    else
+      flash[:error] = "Unable to log in"
+      redirect_to root_path
+    end
+  end
+
+  def show
+    @user = User.find_by(id: params[:id])
+    if @user.nil?
+      head :not_found
+    end
+  end
+
+  def logout
+    user = User.find_by(id: session[:user_id])
```

Comment:
This line gets repeated twice... Maybe this is a good use case for a helper method to be pulled out, or even a controller filter (`before_action`)?